valhalla/midgard/polyline2.h

```diff
@@ -63,6 +64,15 @@ template <class coord_t> class Polyline2 {
    */
   template <class container_t> static typename coord_t::value_type Length(const container_t& pts);
 
+  /**
+   * In an O(n^2) manner (only useful for debugging/testing), checks if there are any
+   * segments in the polyline that are intersecting. If so, returns those intersection
+   * points.
+   * @param   pts  Polyline vertices.
+   * @return  List of intersections found, if any.
+   */
+  std::list<coord_t> GetSelfIntersections();
```

Comment:
Happy to make that change.  

Just fyi, my reason for selecting `list` was that the routine has no idea how many intersections it might return, so being unable to make a guess how much space should be `reserve`ed,  I decided to use `list` which unlike `vector` has low growth cost.